src/cmd/services/m3coordinator/downsample/metrics_appender.go

```diff
@@ -242,13 +244,17 @@ func (a *metricsAppender) SamplesAppender(opts SampleAppenderOptions) (SamplesAp
 	// N.B - it's safe the reuse the shared tag iterator because the matcher uses it immediately to resolve the optional
 	// namespace tag to determine the ruleset for the namespace. then the ruleset matcher reuses the tag iterator for
 	// every match computation.
-	matchResult, err := a.matcher.ForwardMatch(a.tagIter, fromNanos, toNanos, rules.MatchOptions{
+	matchOptions := &rules.MatchOptions{
```

Comment:
why do we need a pointer to the Options? it should be enough for the Options to have a ref to the results.